Userland/Libraries/LibCore/Socket.cpp

```diff
@@ -96,7 +106,13 @@ ErrorOr<void> Socket::connect_local(int fd, ByteString const& path)
 
 ErrorOr<void> Socket::connect_inet(int fd, SocketAddress const& address)
 {
-    auto addr = address.to_sockaddr_in();
+    if (address.type() == SocketAddress::Type::IPv6) {
+        auto addr = address.to_sockaddr_in6();
+
+        return System::connect(fd, bit_cast<struct sockaddr*>(&addr), sizeof(addr));
+    } else {
+        auto addr = address.to_sockaddr_in();
+    }
     return System::connect(fd, bit_cast<struct sockaddr*>(&addr), sizeof(addr));
```

Comment:
Somethings off with the braces here `addr` is out of scope at the last `connect` call
(maybe just elide the `else {}` as it is redundant after a `return`)